stdlib/REPL/src/REPL.jl

```diff
@@ -1274,55 +1274,28 @@ answer_color(r::StreamREPL) = r.answer_color
 input_color(r::LineEditREPL) = r.envcolors ? Base.input_color() : r.input_color
 input_color(r::StreamREPL) = r.input_color
 
+function _without_strings_and_comments(s::AbstractString)
+    # """hello"""
+    s = replace(s, r"\"\"\".*\"\"\"" => "")
```

Comment:
This line deletes the semicolon in
```jl
s = " \"\"\"foo\"\"\" ; # \"\"\" "
```
I'm really skeptical that robust semicolon detection is possible with regex alone.

Reply:
Also, you need the `s` modifier on the regex to match multi-line strings.

But then you run into another corner case.  Consider:
```jl
s = "foo # \"\"\"\n bar; # \"\"\" "
```
Doing `replace(s, r"\"\"\".*?\"\"\""s => "")` deletes the semicolon.